ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -55,26 +96,36 @@ export default function EditGasPopover({
     if (!transaction || !mode) {
       closePopover();
     }
+
+    const newGasSettings =
+      gasEstimateType === GAS_ESTIMATE_TYPES.FEE_MARKET
+        ? {
+            gas: decimalToHex(gasLimit),
+            gasLimit: decimalToHex(gasLimit),
```

Comment:
This seems like a mistake :thinking: 
```suggestion
```

Reply:
Good shout, removed!

Reply:
Need this, its part of the new options bag for speed up and cancel methods, and transactionFactory from ethereumjs/tx looks for gasLimit too. We need this line for now we can figure out if its redundant later.

Reply:
https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/index.js#L557-L625

Reply:
That's disappointing. The type definition even states that there is no gas limit: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/index.js#L56

Reply:
What is the intended difference between `gas` and `gasLimit`? I'm finding this very hard to follow. `generateNewGasParams` for example uses the truthiness of `gasLimit` to decide whether to ensure `gas` was set, but it doesn't use `gasLimit` directly. It doesn't seem to be explained in any comments either :disappointed:

Reply:
A big reason for the introduction of `gasLimit` is because `@ethereumjs/tx` has no mention of a `gas` field in its [`txData` type](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/types.ts#L112-L163), but does specify `gasLimit`.

`generateNewGasParams` has some flaws in its types defined and I did not do a good enough job writing down my internal thoughts/strategy. I had assumed that the usage of `gasLimit` by `@ethereumjs/tx` was a strong signal that we should also be using it instead of `gas` but in the midst of that discovered that the JSON RPC spec does not use `gasLimit`. It doesn't matter because we only use that package to generate a serialized tx that abstracts away field names. 

My thought was to consolidate to using one field name, personally gasLimit makes the most sense to me and is more descriptive than gas, but I wanted to take the time to understand the impact that would have. You'll see in the signTransaction method I added gasLimit to the txParams before signing to match the type. 

All of this to say, I made some choices during individual units of work that led us here. I'll make it a priority to update the documentation of the new method and the create speed up/cancel methods. I don't think this should be addressed here. Happy to have further conversation about this.

Reply:
Hmm. So just to be clear - both properties have the same intended purpose? They're just duplicated for the aforementioned reasons.

I'm asking because your earlier comment made me think that both were needed for speed up and cancellation specifically, but not for other transactions.

Reply:
yes, they are redundant fields, they ultimately set the `gas` value on txParams.